builtin/update-index.c

```diff
@@ -465,6 +473,15 @@ static void update_one(const char *path)
 		fprintf(stderr, "Ignoring path %s\n", path);
 		return;
 	}
+	if (!path_in_cone_modesparse_checkout(path, &the_index)) {
+		/*
+		 * If the path is outside the sparse checkout cone, expand the index.
+		 * Although the path *might* not ultimately be modified (e.g., if no
+		 * flags are set and --no-ignore-skip-worktree-entries is not used), the
+		 * index should be expanded to be safe.
+		 */
+		ensure_full_index(&the_index);
+	}
```

Comment:
This seems like an appropriate protection for now.

Reply:
You advice from earlier applies here - I don't think this check was ultimately needed (thanks to the other implicit index expansion points).